src/id.rs

```diff
@@ -40,9 +40,19 @@ impl FromStr for ResourceId {
     fn from_str(s: &str) -> Result<Self> {
         let (l, r) = s.split_once('-').ok_or(ArklibError::Parse)?;
         let data_size: u64 = l.parse().map_err(|_| ArklibError::Parse)?;
-        let crc32: u32 = r.parse().map_err(|_| ArklibError::Parse)?;
-
-        Ok(ResourceId { data_size, crc32 })
+        let s = r.trim_start_matches('[').trim_end_matches(']');
+        let bytes: Vec<&str> = s.split(", ").collect();
+        if bytes.len() != 32 {
+            return Err(ArklibError::Parse);
+        }
+        let mut blake3 = [0u8; 32];
+        for (i, byte_str) in bytes.iter().enumerate() {
+            blake3[i] = byte_str
+                .trim()
+                .parse()
+                .map_err(|_e| ArklibError::Parse)?;
+        }
```

Comment:
Something like https://crates.io/crates/base64 should be used instead of `{:?}`.